src/gen_caml.ml

```diff
@@ -234,16 +235,23 @@ let output_params_binder _ vars =
   output "let set_params stmt =";
   inc_indent ();
   output "let p = T.start_params stmt (%s) in" (eval_count_params vars);
-  List.iteri set_var vars;
+  List.iteri (fun index v -> set_var index v) vars;
   output "T.finish_params p";
   dec_indent ();
   output "in";
   "set_params"
 
+let exclude_in_vars l =
+  List.filter
+    (function
+      | Single _ | Choice _ -> true
```

Comment:
this is interesting, observations:
1. choice should be recursed
2. choice itself is also text-time substitution, so by itself it should lead to T.no_params but in majority of cases the branches will have internal parameters
3. i wonder if it is only an issue of pretty output but not correctness because output_params_binder with empty list as input will just  generate identity function